### ISSUE TITLE
pipenv picks up anaconda defunct installation and crashes with invalid python path.

### ISSUE THREAD

**Issue by rhettprince** [pypa/pipenv#4486]
I uninstalled anaconda. and only use python 3.8. 
pipenv picks up the defunct anaconda paths from I have no idea where and crashes whenit determins it sia an invalid python path. 

Be sure to check the existing issues (both open and closed!), and make sure you are running the latest version of Pipenv.

Check the [diagnose documentation](https://pipenv.pypa.io/en/latest/diagnose/) for common issues before posting! We may close your issue if it is very similar to one of them. Please be considerate, or be on your way.

Make sure to mention your debugging experience if the documented solution failed.

### Issue description

Describe the issue briefly here.

### Expected result

Describe what you expected.

### Actual result

When possible, provide the verbose output (`--verbose`), especially for locking and dependencies resolving issues.

### Steps to replicate

Provide the steps to replicate (which usually at least includes the commands and the Pipfile).

-------------------------------------------------------------------------------

Please run `$ pipenv --support`, and paste the results here. Don't put backticks (`` ` ``) around it! The output already contains Markdown formatting.

If you're on macOS, run the following:

    $ pipenv --support | pbcopy

If you're on Windows, run the following:

    > pipenv --support | clip

If you're on Linux, run the following:

    $ pipenv --support | xclip
10:31 AM 9/29/2020
Microsoft Windows [Version 6.3.9600]
(c) 2013 Microsoft Corporation. All rights reserved.

C:\ddev>pipenv shell
Traceback (most recent call last):
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\python.py", line 618, in parse_executable
    result_version = get_python_version(path)
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\utils.py", line 105, in get_python_version
    c = subprocess.Popen(version_cmd, **subprocess_kwargs)
  File "c:\program files (x86)\python38-32\lib\subprocess.py", line 854, in __in
it__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "c:\program files (x86)\python38-32\lib\subprocess.py", line 1307, in _ex
ecute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\program files (x86)\python38-32\lib\runpy.py", line 194, in _run_modu
le_as_main
    return _run_code(code, main_globals, None,
  File "c:\program files (x86)\python38-32\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Program Files (x86)\Python38-32\Scripts\pipenv.exe\__main__.py", line
 7, in <module>
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\click
\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\click
\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\click
\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\click
\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\click
\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\click
\decorators.py", line 73, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\click
\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\cli\command.
py", line 428, in shell
    do_shell(
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\core.py", li
ne 2356, in do_shell
    ensure_project(
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\core.py", li
ne 576, in ensure_project
    ensure_virtualenv(
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\core.py", li
ne 498, in ensure_virtualenv
    python = ensure_python(three=three, python=python)
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\core.py", li
ne 388, in ensure_python
    path_to_python = find_a_system_python(python)
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\core.py", li
ne 350, in find_a_system_python
    return next(iter(finder.find_all_python_versions()), None)
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\vendor\pytho
nfinder\pythonfinder.py", line 312, in find_all_python_versions
    python_version_dict = getattr(self.system_path, "python_version_dict", {})
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\vendor\pytho
nfinder\pythonfinder.py", line 120, in system_path
    self._system_path = self.create_system_path()
  File "c:\program files (x86)\python38-32\lib\site-packages\pipenv\vendor\pytho
nfinder\pythonfinder.py", line 82, in create_system_path
    return pyfinder_path.SystemPath.create(
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\path.py", line 682, in create
    instance = instance._run_setup()
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\path.py", line 233, in _run_setup
    new_instance = new_instance._setup_windows()
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\path.py", line 411, in _setup_windows
    windows_finder = WindowsFinder.create()
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\windows.py", line 146, in create
    return cls()
  File "<attrs generated init pythonfinder.models.windows.WindowsFinder>", line
13, in __init__
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\windows.py", line 113, in get_versions
    versions[py_version.version_tuple[:5]] = base_dir
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\python.py", line 365, in __getattribute__
    result = super(PythonVersion, self).__getattribute__(key)
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\python.py", line 430, in version_tuple
    self.patch,
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\python.py", line 375, in __getattribute__
    instance_dict = self.parse_executable(executable)
  File "C:\Program Files (x86)\Python38-32\Lib\site-packages\pipenv\vendor\pytho
nfinder\models\python.py", line 620, in parse_executable
    raise ValueError("Not a valid python path: %r" % path)
ValueError: Not a valid python path: 'C:/ProgramData/Anaconda3/python.exe'

C:\ddev>

**Comment by saitejaswi18** [comment]
how did u rectify that. I am facing the same issue

**Comment by rhettprince** [comment]
hi, 

 

I looked back throughmy posts and somehow I found a reply to my post. 

the gentleman said he had to return to an earlier version of pipenv 

 

here is what I copied from his solution and it worked 

 

 

quote from his post 

 

Using 

pip install pipenv==2018.10.13 

like Wyzzard123 mentioned fixed the problem for me. I was stuck in Anaconda3 not being a vlid python path before

 

end of quoted 

 

I pip uninstall pipenv and 

pip install his version. 

 

pipenv seems to work now. 

I am concerned that pipenv is getting worse not better.
